### DISCUSSION TITLE
[Bug]:

### DISCUSSION THREAD

**Discussion by Teddy937** [spatie/laravel-pdf#55] [Q&A]
### What happened?

Laravel spatie pdf works well on Mac local env but upon deploying on Ubuntu server am getting some errors.

### How to reproduce the bug

When I hit the endpoint to generate pdf I see the error below

Symfony \ Component \ Process \ Exception \ ProcessFailedException
The command "PATH=$PATH:/usr/local/bin:/opt/homebrew/bin NODE_PATH=`npm root -g` node '/var/www/experts/backend/vendor/spatie/browsershot/src/../bin/browser.cjs' '{"url":"file:\/\/\/tmp\/512163753-0320285001705991912\/index.html","action":"pdf","options":{"args":[],"viewport":{"width":800,"height":600},"displayHeaderFooter":false,"printBackground":true}}'" failed. Exit Code: 1(General error) Working directory: /var/www/experts/backend/public Output: ================ Error Output: ================ [1m[43m[30m Puppeteer old Headless deprecation warning:[0m[33m In the near future `headless: true` will default to the new Headless mode for Chrome instead of the old Headless implementation. For more information, please see https://developer.chrome.com/articles/new-headless/. Consider opting in early by passing `headless: "new"` to `puppeteer.launch()` If you encounter any bugs, please report them to https://github.com/puppeteer/puppeteer/issues/new/choose.[0m Error: Could not find Chrome (ver. 119.0.6045.105). This can occur if either 1. you did not perform an installation before running the script (e.g. `npx puppeteer browsers install chrome`) or 2. your cache path is incorrectly configured (which is: /root/.cache/puppeteer). For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration. at ChromeLauncher.resolveExecutablePath (/var/www/experts/backend/node_modules/puppeteer/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ProductLauncher.js:278:27) at ChromeLauncher.executablePath (/var/www/experts/backend/node_modules/puppeteer/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ChromeLauncher.js:209:25) at ChromeLauncher.computeLaunchArguments (/var/www/experts/backend/node_modules/puppeteer/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ChromeLauncher.js:103:37) at async ChromeLauncher.launch (/var/www/experts/backend/node_modules/puppeteer/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ProductLauncher.js:69:28) at async callChrome (/var/www/experts/backend/vendor/spatie/browsershot/bin/browser.cjs:91:23)

### Package Version

spatie/laravel-pdf version 1.1

### PHP Version

8.1.27

### Laravel Version

10.10

### Which operating systems does with happen with?

Linux

### Notes

This works well when developing on macOs locally. But fails on Ubuntu server.

**Comment by Teddy937** [comment]
My cache path is  /root/.cache/puppeteer

**Reply by KS-Verbosity** [reply]
Any resolution available for this? I have set up a Forge server and have encountered the same issue. Many attempts to solve it have failed so far.

**Comment by MichaelCorrea478** [comment]
Solved the problem following these steps:
1 - create the folder `.cache/puppeteer` in the root of the project
2 - create a config file `.puppeteerrc.cjs` in the root of the project with the following content: 
```
const {join} = require('path');

/**
 * @type {import("puppeteer").Configuration}
 */
module.exports = {
  // Changes the cache location for Puppeteer.
  cacheDirectory: join(__dirname, '.cache', 'puppeteer'),
};
```
See Puppeteer [documentation](https://pptr.dev/guides/configuration)
3 - Uninstall puppeteer: `npm uninstall puppeteer`
4 - Remove cache folder: `rm -rf ~/.cache/puppeteer`
5 - Install puppeteer again: `npm install puppeteer`

This worked for me.

#### PHP Version
8.2.7
#### Laravel Version
10.10
#### spatie/laravel-pdf Version
1.5
#### Operating system
Linux - CentOS Stream 9

**Reply by BinganaDev** [reply]
it works for me also, big thannnnnnnnnkkk you....

**Comment by ArielMejiaDev** [comment]
@MichaelCorrea478 when the installation process finish the file `.puppeteerrc.cjs` should still be present?

the same file should be added in every project that uses puppeteer?

Thanks

**Reply by MichaelCorrea478** [reply]
As far as I know, the `.pupeteerrc.cjs` file should remain. Each project has to have its own config file, but it does not need to point to the project root directory. You can create a folder outside your project, anywhere in your system and make each config file point to it.

I choose to keep this folder inside the project folder because I have just one project running in this production environment, and didn't want to deal with permission issues.
